screen2.0/src/app/search/page.tsx

```diff
@@ -1,91 +1,517 @@
-    prim_e: searchParams.prim_e ? Number(searchParams.prim_e) : 2,
-    mamm_s: searchParams.mamm_s ? Number(searchParams.mamm_s) : -4,
-    mamm_e: searchParams.mamm_e ? Number(searchParams.mamm_e) : 8,
-    vert_s: searchParams.vert_s ? Number(searchParams.vert_s) : -3,
-    vert_e: searchParams.vert_e ? Number(searchParams.vert_e) : 8,
-    //Linked Genes Filter
-    genesToFind: searchParams.genesToFind ? searchParams.genesToFind.split(",") : null,
-    distancePC: searchParams.distancePC ? checkTrueFalse(searchParams.distancePC) : true,
-    distanceAll: searchParams.distanceAll ? checkTrueFalse(searchParams.distanceAll) : true,
-    distanceFromcCRE: searchParams.distanceFromcCRE ? Number(searchParams.distanceFromcCRE) : 1000000,
-    CTCF_ChIA_PET: searchParams.CTCF_ChIA_PET ? checkTrueFalse(searchParams.CTCF_ChIA_PET) : true,
-    RNAPII_ChIA_PET: searchParams.RNAPII_ChIA_PET ? checkTrueFalse(searchParams.RNAPII_ChIA_PET) : true
+      else setPage(page - 1)
+    }
+    //No action needed when closing a tab to the right of the page you're on
+  }
+
+  const handlePageChange = (_, newValue: number) => {
+    if (mainQueryParams.gene.name && (newValue === 2 || newValue === 3)) {
+      setOpen(false)
+    }
+    setPage(newValue)
   }
 
-  //Contains cell type data of the specified assembly
-  const globals: CellTypeData = await getGlobals(mainQueryParams.assembly)
+  //Keep URL and state in sync. Prevent from firing initially to allow time for opencCREs to be initialized
+  useEffect(() => {
+    //Check if the URL params representing state are stale
+    if (
+      opencCREsInitialized &&
+      JSON.stringify(constructMainQueryParamsFromURL(searchParams)) !== JSON.stringify(mainQueryParams)
+      || JSON.stringify(constructFilterCriteriaFromURL(searchParams)) !== JSON.stringify(filterCriteria)
+      || JSON.stringify(constructBiosampleTableFiltersFromURL(searchParams)) !== JSON.stringify(biosampleTableFilters)
+      || +searchParams.page !== page
+      || searchParams.accessions !== opencCREs.map(x => x.ID).join(','))
+     {
+      const newURL = constructSearchURL(
+        mainQueryParams,
+        filterCriteria,
+        biosampleTableFilters,
+        page,
+        opencCREs.map(x => x.ID).join(',')
+      )
+      router.push(newURL)
+    }
+  }, [searchParams, mainQueryParams, filterCriteria, biosampleTableFilters, page, opencCREs, router, basePathname, opencCREsInitialized])
+
+  //fetch globals
+  useEffect(() => {
+    // Setting react/experimental in types is not fixing this error? https://github.com/vercel/next.js/issues/49420#issuecomment-1537794691
+    // @ts-expect-error
+    startTransition(async () => {
+      console.log("fetching globals")
+      setGlobals(await getGlobals(mainQueryParams.coordinates.assembly))
+    })
+  }, [mainQueryParams.coordinates.assembly])
+
+
+  //Fetch raw cCRE data (main query and linked genes)
+  useEffect(() => {
+      setLoadingFetch(true)
+
+      let start: number;
```

Comment:
we can just assign start with mainQueryParams.coordinates.start initially and else block can be removed

Reply:
fixed